src/library/tactic/simp_lemmas.cpp

```diff
@@ -870,20 +870,25 @@ static simp_lemmas add_core(type_context_old & ctx, simp_lemmas const & s, name
             proof = mk_app(proof, mvar);
         }
         expr lhs, rhs;
+        name head_name;
         if (is_iff(type, lhs, rhs)) {
-            // We need to turn the `iff` into a `eq` for `simp`. `dsimp` doesn't look at this proof,
-            // but does require that we use `get_eq_name()` below and not `get_iff_name()`.
-            proof = mk_propext(lhs, rhs, proof);
```

Comment:
Is this proof not used, even for `simp`?

Reply:
The proof is used by `simp`; the new diff just doesn't invoke `propext` here, as its best to let `simp` handle that itself so that this lemma is keyed by the right head symbol.